### DISCUSSION TITLE
Helper Extension Methods for common Scenarios

### DISCUSSION THREAD

**Discussion by TonyValenti** [oqtane/oqtane.framework#808] [General]
@sbwalker -
Are you open to extension methods being added to Oqtane that augment default namespaces?

For example, I add this code to every project I do:
```
namespace System.Collections.Generic {
    public static class EnumerableExtensions {

        public static IEnumerable<T> EmptyIfNull<T>(this IEnumerable<T>? source) {
            return source ?? Enumerable.Empty<T>();
        }

       //There is more but I'm omitting it
   }
}
```

So that instead of this:
```
            if (TeamManagers != null)
            {
                foreach(var TeamManager in TeamManagers)
                {
                    _TeamManagerRepository.AddTeamManager(new Models.TeamManager { ModuleId = module.ModuleId, Name = TeamManager.Name });
                }
            }
```

I can just do this:
```
                foreach(var TeamManager in TeamManagers.EmptyIfNull())
                {
                    _TeamManagerRepository.AddTeamManager(new Models.TeamManager { ModuleId = module.ModuleId, Name = TeamManager.Name });
                }
```

Other helper methods I use are ones like:
```
        public static bool IsNullOrEmpty(this string? This) {
            return string.IsNullOrEmpty(This);
        }

        public static bool IsNotNullOrEmpty(this string? This) {
            return !string.IsNullOrEmpty(This);
        }

        public static bool IsBlank(this string? This) {
            return string.IsNullOrWhiteSpace(This);
        }

        public static bool IsNotBlank(this string? This) {
            return !string.IsNullOrWhiteSpace(This);
        }

        public static string Join(this IEnumerable<string> This, string Separator) {

            var ret = string.Join(Separator, This);

            return ret;
        }
```

Thoughts?

**Comment by bryfa** [comment]
In my opinion TeamManagers should not be null in the first place. So I think it would be better to just adopt the nullable checks with C# 8. So that the compiler can warn you that this maybe null. And do a proper fix for that by initialising the variable with an empty list. Then you don't have to implement the test and the extension method at all.

**Reply by bryfa** [reply]
As an alternative for your solution on EmptyIfNull look at this article:
https://www.enterprise-software-development.eu/posts/2019/11/19/option-type.html

**Comment by hishamco** [comment]
@TonyValenti fortunately, I added this few days ago ;)

https://github.com/oqtane/oqtane.framework/blob/81475fd83516ca7c723d967f58d278a22e6f050a/Oqtane.Shared/Extensions/EnumerableExtensions.cs#L5-L6
